### PR TITLE
[clang] Fix compile-time regression from attribute arg checking change

### PR DESCRIPTION
**Pull request by mikerice1969**
In 2acf77f987331c05520c5bfd849326909ffce983 code was added to use the 'full' name including syntax and scope.

Instead of building up a large string for each name, add syntax and scope checks to the value expression in tablegen.

There is already code to generate expressions for target specific attributes. This change refactors and adds to that code to include syntax and scope checks.

The tablegen avoids generating the complicated expression unless there are two attributes using the same name, otherwise the case values will be as simple as before.

Removes the currently unused attributeHasStrictIdentifierArgAtIndex function and the related tablegen.

### CODE REVIEW

**Review by AaronBallman — APPROVED** [review]
The code changes LGTM but I wonder if we should put a branch up on https://llvm-compile-time-tracker.com/ to verify the compile-time improvements before landing to verify they in fact resolve the issue.

**Comment by nikic** [comment]
Compile-time: https://llvm-compile-time-tracker.com/compare.php?from=cc25748d07b0adb1a0223e836e9cae22af150496&to=f5da93cb88d483ff0fa01385579c0079d375fb15&stat=instructions:u

Original regression for comparison: https://llvm-compile-time-tracker.com/compare.php?from=39b6900852e7a1187bd742ba5c1387ca1be58e2c&to=2acf77f987331c05520c5bfd849326909ffce983&stat=instructions:u

Looks about the same!

**Comment by mikerice1969** [comment]
Thanks @nikic!

**Pull request merged**
